### ISSUE TITLE
Vector3 expression node

### ISSUE THREAD

**Issue by BastiaanOlij** [BastiaanOlij/gdprocmesh#17]
Hey @Mr-Slurpy , 

Got to play with your expression node finally today. Love it!!
I think we should move it into the transform group instead of the primitive group. It is a node that performs a modification on an input array.

The other thing I was wondering about, and this may be more difficult, but it would be nice if it could have a Vertex3 as an output, or maybe have a version that has a Vertex3 as output. It'll need a larger expression field as it will end up being something like "Vector3(a, sin(b), c)" as an expression (that happened to be what I was using it for).

**Comment by Mr-Slurpy** [comment]
Yeah sure we could add a new node for Vector3 expressions. I was thinking of trying out C++ templates to cut down on code.

For the group, I agree maybe it belongs in the the transform group. Also we shouldn't let the transform group get too big. (i think 7-8 options max) If the transform group gets too big we should move some nodes out or make another submenu in it. For example,
![proc_menus](https://user-images.githubusercontent.com/15792219/44642416-080e1b00-a99a-11e8-8787-5485e4291aac.PNG)

P.S. We should do something about all the gdscript warnings popping up in the console. If thats even possible.
![proc_warnings](https://user-images.githubusercontent.com/15792219/44642293-8d450000-a999-11e8-992f-cdd84c9cc34a.PNG)

**Comment by BastiaanOlij** [comment]
I'm not sure if template classes will really prevent that much code duplication, but it is worth a try. 

I agree adding one more subgroup makes sense, we may eventually need to move to a different menu structure all together. The menu was just the quickest thing I could come up with but I'm still learning about Godots UI :)

I'm not sure if we can do much about the GDScript warnings as atleast 3 of those 4 are due to parameters of signals we're simply ignoring. That first one is probably a leftover from before I split off the node code into a separate script so that can be fixed.
